xl2times/datatypes.py

```diff
@@ -70,6 +71,55 @@ def has_tag(cls, tag):
         return tag in cls._value2member_map_
 
 
+class DataModule(str, Enum):
+    """Categorise data into modules based on the file they are coming from."""
+
+    base = "VT_*.*, BY_Trans.*"
+    syssettings = "SysSettings.*"
+    subres = "SubRES_TMPL/SubRES_*.*"
+    sets = "Set*.*"
+    lma = "LMA*.*"
+    demand = "SuppXLS/Demands/Dem_Alloc+Series.*, SuppXLS/Demands/ScenDem_*.*"
+    scen = "SuppXLS/Scen_*.*"
+    trade = "SuppXLS/Trades/ScenTrade_*.*"
+
+    @classmethod
+    def is_module_type(cls, path):
+        for data_module in cls:
+            if any(
+                PurePath(path.lower()).match(pattern.lower().strip())
+                for pattern in data_module.value.split(",")
+            ):
+                return data_module
+        return None
+
+    @classmethod
+    def is_submodule_type(cls, path):
+        match cls.is_module_type(path):
+            case DataModule.base | DataModule.subres:
+                if PurePath(path.lower()).match("*_trans.*"):
+                    return "trans"
+                else:
+                    return "main"
+            case None:
+                return None
+            case _:
+                return "main"
+
+    @classmethod
+    def is_module_name(cls, path):
+        module_type = cls.is_module_type(path)
+        match module_type:
+            case DataModule.base | DataModule.sets | DataModule.lma | DataModule.demand | DataModule.trade | DataModule.syssettings:
+                return module_type.name
+            case DataModule.subres:
+                return re.sub("_trans$", "", PurePath(path).stem.lower())
+            case None:
+                return None
+            case _:
```

Comment:
If this case is meant to catch the `scen` case, it's better to change it to be explicitly `case scen` as then we'll get a more understandable error if we in the future expand the number of Enums in `DataModule`, instead of it silently returning the path's stem and potentially throwing an error down the line.

Reply:
Currently it is just `scen`, but I think we'll need to remove some from the firsth `case`, so there will be others. As long as it is a `DataModule` returning path's stem here should be a good general approach.

Reply:
Perhaps it's not so relevant to this instance, but in general when using case statements it's good programming practice to avoid using `_` as much as possible and to explicitly enumerate cases. That way the type checker / compiler will warn you of all the places where you need to check that the case logic still holds when you add a new case to the Enum. So the little bit of extra effort of updating this catch-all case will hopefully be worth it if in the future we add a new case to the Enum for which returning stems was actually not the correct logic.

But I'll defer to you on whether this makes sense to do here. :)

Reply:
Definitely agree that catch all cases should be avoided! In this case, I felt it was ok, because we make sure that all non-`None` values are valid.
Let's see how to change this part is more developed.